dali/kernels/CMakeLists.txt

```diff
@@ -38,7 +38,12 @@ adjust_source_file_language_property("${DALI_KERNEL_SRCS}")
 set_source_files_properties("dummy.cu" PROPERTIES LANGUAGE CUDA)
 add_library(dali_kernels ${LIBTYPE} ${DALI_KERNEL_SRCS})
 target_link_libraries(dali_kernels PUBLIC dali_core)
-target_link_libraries(dali_kernels PRIVATE ${CUDA_cufft_static_LIBRARY})
+if (WITH_DYNAMIC_CUDA_LIBS)
+  target_link_libraries(dali_kernels PRIVATE dynlink_cufft)
+else()
+  target_link_libraries(dali_kernels PRIVATE ${CUDA_cufft_LIBRARY})
+  target_link_libraries(dali_kernels PRIVATE "-Wl,--exclude-libs,${CUDA_cufft_LIBRARY}")
```

Comment:
I'm a little bit lost at these 2 lines. Firstly we link `CUDA_cufft_LIBRARY` and then we link it, passing `exclude-libs` option to the linker?

Reply:
_Specifies a list of archive libraries from which symbols should not be automatically exported. The library names may be delimited by commas or colons. Specifying --exclude-libs ALL excludes symbols in all archive libraries from automatic export. This option is available only for the i386 PE targeted port of the linker and for ELF targeted ports. For i386 PE, symbols explicitly listed in a .def file are still exported, regardless of this option. For ELF targeted ports, symbols affected by this option will be treated as hidden._